governance.md

```diff
@@ -59,260 +70,170 @@ The foundations of Project governance are:
-participation, community help/building, education and outreach, design work,
-etc. We are deliberately not setting arbitrary quantitative metrics (like “100
-commits in this repo”) to avoid encouraging behavior that plays to the metrics
-rather than the project’s overall well-being. We want to encourage a diverse
-array of backgrounds, viewpoints and talents in our team, which is why we
-explicitly do not define code as the sole metric on which Core Team membership
-will be evaluated.
-
-If a Core Team member becomes inactive in the project for a period of one year,
-they will be considered for removal from the Core Team. Before removal,
-inactive Member will be approached by the BDFL to see if they plan on returning
-to active participation. If not they will be removed immediately upon a Core
-Team vote. If they plan on returning to active participation soon, they will be
-given a grace period of one year. If they don't return to active participation
-within that time period they will be removed by vote of the Core Team without
-further grace period. All former Core Team members can be considered for
-membership again at any time in the future, like any other Project Contributor.
-Retired Core Team members will be listed on the project website, acknowledging
-the period during which they were active in the Core Team.
-
-The Core Team reserves the right to eject current Members, other than the BDFL,
-if they are deemed to be actively harmful to the project’s well-being, and
-attempts at communication and conflict resolution have failed.
+To manage the project, there are different Teams that each have responsibility for
+specific aspects of the project. Collectively, the members of all Teams are referred to
+as _Stewards_ of the project.
+
+## Steering Committee
+
+The role of the _pandas_ Steering Committee is to coordinate the activities of the
+different Teams and to ensure that different policies and procedures are carried out in
+a consistent manner. The Steering Committee has no routine decision-making authority,
+except as detailed herein, although in exceptional circumstances it may be called upon
+from time to time to make decisions that are in the best interest of The Project as a
+whole. The Steering Committee will itself decide when a circumstance is exceptional.
+
+The Steering Committee may create a working group to consider changes to the governance
+model described in this document, including the creation of new Teams that support the
+goals of The Project. Changes to the governance model will require a voting process and
+approval equivalent to the process described in PDEP-1, with the exception that all
+Stewards are eligible to vote.  
+
+The Steering Committee may appoint temporary working groups to work on issues, such as
+governance, that fall outside the scope of responsibilities for the existing Teams.
+
+The ideal composition of the Steering Committee consists of 5 people.  The Steering
```

Comment:
The current number of 5 was a compromise between the people involved getting to this initial version. Personally I think to also like a smaller team of 3 (and brought this up before in the meeting), but others felt that this was too small of a team. 

It's of course a project of a completely different size and scope, so difficult to compare, but for example the Python steering council also consist of 5 people, and that doesn't seem to prevent them from making decisions (my impression from the sideline).

Reply:
It could be that each steering group member represents one of the teams and is accountable to that team. Has this been considered?

Each team could have a deputy representative to participate in the discussions/votes when the steering committee member is unavailable/on holiday.

The teams could be responsible for choosing/electing their representative on the steering council.

I'm not sure if this would increase or decrease the efficiency of making decisions but would maybe make the decisions focused on team goals and not on individual preferences?

Of course the upside/downside (depending on viewpoint) of this would mean that each team has equal importance on decision making.

Reply:
I didn't spot anywhere in this document that says a steward can only be a member of one team. I have assumed that one may be in multiple teams and this is is probably likely for most people is some specific teams. Not sure how this would tally with the above idea, but possibly the people who are in multiple teams are the ones most interested / qualified in seeking votes to the steering committee anyway?

Reply:
Actually the teams document has criterias of membership being in other teams, so this is implicit.

Reply:
Yes, I suspect that as most teams will have members that are also in the core library team, that the representatives would also be from the core library team.

The difference that I see would be that the steering committee would have those core library members being accountable to different aspects of the project and not just a superset of the core library team (even though the members would probably be the same individuals).

Reply:
> It could be that each steering group member represents one of the teams and is accountable to that team. Has this been considered?

We did discuss this, but felt that would make the Steering Committee too large, especially if we decide to add more teams.

Reply:
> Actually the teams document has criterias of membership being in other teams, so this is implicit.

Yes, although I think it is worth making this more explicit in this document.

Reply:
I agree that reducing the responsibilities of the core (library) team is a great way to allow the core developers to concentrate on the issues that interest them the most.

(I personally had no interest in CI/infrastructure but this was a necessary evil while managing the releases)

I think that the initial selection for the teams should be any current core developer that has an interest in that area and allow the core developers to choose themselves which responsibilities they are happy to relinquish.

( I personally would be happy not to be on any of the teams. I consider my presence on the pandas-stubs team as purely a representative of the core team from the perspective of maintaining consistency with the in-line types)

But that leaves the steering committee. Now it appears that this is intended to replace the BDFL. Now Wes is not usually called upon to make decisions and we now have the PDEP voting process for technical decisions. I would hope that the steering committee does not have the power to override a PDEP vote and so that leaves very little for the steering committee to do. (I am concerned with the text _The Steering Committee will itself decide when a circumstance is exceptional_ as this gives the small group carte blanche to override a PDEP decision.)

Now the steering committee is defined as

_The role of the _pandas_ Steering Committee is to coordinate the activities of the
different Teams and to ensure that different policies and procedures are carried out in
a consistent manner._

and so to me it makes sense that those teams should somehow be represented.

Reply:
> 
> I think that the initial selection for the teams should be any current core developer that has an interest in that area and allow the core developers to choose themselves which responsibilities they are happy to relinquish.

Yes, that is the plan.

> 
> But that leaves the steering committee. Now it appears that this is intended to replace the BDFL. Now Wes is not usually called upon to make decisions and we now have the PDEP voting process for technical decisions. I would hope that the steering committee does not have the power to override a PDEP vote and so that leaves very little for the steering committee to do. (I am concerned with the text _The Steering Committee will itself decide when a circumstance is exceptional_ as this gives the small group carte blanche to override a PDEP decision.)

I think we should say that the Steering Committee cannot override a PDEP.

> 
> Now the steering committee is defined as
> 
> _The role of the _pandas_ Steering Committee is to coordinate the activities of the different Teams and to ensure that different policies and procedures are carried out in a consistent manner._
> 
> and so to me it makes sense that those teams should somehow be represented.

An earlier draft version of the proposal that we discussed in the governance meetings did say that each team would have a representative on the Steering Committee, but then it was felt that it would be giving equal weight to all of the teams, and would also make the committee too large.  Maybe we should consider that certain teams must have someone on the Steering Committee, but one person can represent multiple such teams.  (E.g., someone on docs and core library could represent both).   The concern with doing that is that some of the teams could be small, so that restricts who can be on the Steering Committee.

Our thinking evolved that if a Team is not represented on the Steering Committee, that when the Steering Committee meets, it should reach out to those non-represented Teams to find out if there are any issues from their team that require coordination across the teams.

Reply:
> Or conversely, that a team without any members on the steering committee, has one of the steering committee members to represent them and liaise with.

We can be more explicit about this concept in the next revision

Reply:
Added a sentence in the next revision.